enhancements/oc/top-pvc-usage.md

```diff
@@ -0,0 +1,117 @@
+---
+title: top-pvc-usage
+authors:
+- "@gmeghnag"
+- "@tsmetana"
+reviewers:
+- "@dobsonj"
+- "@ardaguclu"
+approvers:
+- "@ardaguclu"
+creation-date: 2024-10-23
+last-updated: 2024-10-23
+status:
+see-also:
+replaces:
+superseded-by:
+---
+
+# top pvc
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Provide a simple command to display a PersistentVolumeClaim capacity usage.
+
+## Motivation
+
+There already exist options like `oc adm top pod` that can display usage
+statistic of the server resources. Adding an option to display also usage
+of the PersistentVolumeClaims seems like a natural extension which provides
+important information about the persistent storage available to the workloads.
+
+The volume usage statistics are already available via the web console but we
+should also have a way to get this through the CLI too.
+
+### Goals
+
+1. Provide a simple CLI option to display filesystem usage of
+   PersistentVolumeClaims.
+2. Display only the percentual usage for a given PersistentVolumeClaim or all
```

Comment:
If user invokes this command with `--all-namespaces`, can we get all the required data in 1 prometheus query? or are we going to send multiple queries per each namespace?. If user has 100 namespaces, sending 100 queries to metrics server would be impractical.

Reply:
- _"If user invokes this command with `--all-namespaces`, can we get all the required data in 1 prometheus query?"_
  Yes, and this is how it works when you execute the command with `--all-namespaces`, a single query will be executed: 
  ```
  100*kubelet_volume_stats_used_bytes{persistentvolumeclaim=~".*"}/kubelet_volume_stats_capacity_bytes{persistentvolumeclaim=~".*"}
  ```

- _"or are we going to send multiple queries per each namespace?"_
  No, this way would overload the metric server, and as you said would be impractical.